udata/core/dataset/models.py

```diff
@@ -290,11 +290,19 @@ class ResourceMixin(object):
     harvest = db.EmbeddedDocumentField(HarvestResourceMetadata)
     schema = db.DictField()
 
-    created_at = db.DateTimeField(default=datetime.now, required=True)
-    modified = db.DateTimeField(default=datetime.now, required=True)
+    created_at_internal = db.DateTimeField(default=datetime.now, required=True)
+    last_modified_internal = db.DateTimeField(default=datetime.now, required=True)
     published = db.DateTimeField()  # DEPRECATED BUT LEFT FOR BACKWARDS COMPATIBILITY
     deleted = db.DateTimeField()
 
+    @property
+    def created_at(self):
+        return self.harvest.created_at if self.harvest and self.harvest.created_at else self.created_at_internal
+
+    @property
+    def last_modified(self):
+        return max([self.last_modified_internal, to_naive_datetime(self.harvest.modified_at)]) if self.harvest and self.harvest.modified_at else self.last_modified_internal
```

Comment:
If you wanna, we can update `harvest.modified_at` -> `harvest.last_modified` for coherence, even if you like the first better (optional)?